src/platform/forms-system/src/js/types.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * @file Generic types for the form system
+ *
+ * Used for intellesense (autocompletion)
+ */
+
+// Example 1:
+// /** @type {FormConfig} */
+// const formConfig = {...};
+
+// Example 2:
+// /** @type {PageSchema} */
+// export default {
+//   uiSchema: {...},
+//   schema: {...},
+// };
+
+// Use @typedef ({{}}) format to use [key: string]: Type
+// Use @typedef {Object} format to be able to add comments to properties
+
+/**
+ * @typedef {Object} FormConfig
+ * @property {Array<Object>} additionalRoutes
```

Comment:
Added new key recommendations, and refactored to this syntax.
This syntax allows for adding "- description". We can add these in future if we want.